.github/workflows/spec-parser.yml

```diff
@@ -1,6 +1,11 @@
 name: run spec-parser
 
-on: [push]
+on:
+  push:
+    branches:
+      - main
```

Comment:
since we are developing on feature branches. It might also be helpful to also run on pushes to them. But we can decide that later and should merge this first.